examples/tensorflow/text-classification/run_glue.py

```diff
@@ -0,0 +1,554 @@
+            )
+    # endregion
+
+    # region Logging
+    logging.basicConfig(
+        format="%(asctime)s - %(levelname)s - %(name)s -   %(message)s",
+        datefmt="%m/%d/%Y %H:%M:%S",
+        handlers=[logging.StreamHandler(sys.stdout)],
+    )
+    logger.setLevel(logging.INFO if is_main_process(training_args.local_rank) else logging.WARN)
+
+    # Set the verbosity to info of the Transformers logger (on main process only):
+    if is_main_process(training_args.local_rank):
+        transformers.utils.logging.set_verbosity_info()
+        transformers.utils.logging.enable_default_handler()
+        transformers.utils.logging.enable_explicit_format()
+    logger.info(f"Training/evaluation parameters {training_args}")
+    # endregion
+
+    # region Dataset and labels
+    # Set seed before initializing model.
+    set_seed(training_args.seed)
+
+    # Downloading and loading a dataset from the hub. In distributed training, the load_dataset function guarantee
+    # that only one local process can concurrently download the dataset.
+    datasets = load_dataset("glue", data_args.task_name, cache_dir=model_args.cache_dir)
+    # See more about loading any type of standard or custom dataset at
+    # https://huggingface.co/docs/datasets/loading_datasets.html.
+
+    is_regression = data_args.task_name == "stsb"
+    if not is_regression:
+        label_list = datasets["train"].features["label"].names
+        num_labels = len(label_list)
+    else:
+        num_labels = 1
+
+    if data_args.predict_file is not None:
+        logger.info("Preparing user-supplied file for predictions...")
+
+        data_files = {"data": data_args.predict_file}
+
+        for key in data_files.keys():
+            logger.info(f"Loading a local file for {key}: {data_files[key]}")
+
+        if data_args.predict_file.endswith(".csv"):
+            # Loading a dataset from local csv files
+            user_dataset = load_dataset("csv", data_files=data_files, cache_dir=model_args.cache_dir)
+        else:
+            # Loading a dataset from local json files
+            user_dataset = load_dataset("json", data_files=data_files, cache_dir=model_args.cache_dir)
+        needed_keys = task_to_keys[data_args.task_name]
+        for key in needed_keys:
+            assert key in user_dataset["data"].features, f"Your supplied predict_file is missing the {key} key!"
+        datasets["user_data"] = user_dataset["data"]
+    # endregion
+
+    # region Load model config and tokenizer
+    #
+    # In distributed training, the .from_pretrained methods guarantee that only one local process can concurrently
+    # download model & vocab.
+    config = AutoConfig.from_pretrained(
+        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+        num_labels=num_labels,
+        finetuning_task=data_args.task_name,
+        cache_dir=model_args.cache_dir,
+        revision=model_args.model_revision,
```

Comment:
We should set the label2id and id2label here as well (so that when finetuned the model has the proper labels on the inference API).

(This is also needed on the PyTorch side, will make a PR soon-ish)